### DISCUSSION TITLE
Global users in multi-tenant application

### DISCUSSION THREAD

**Discussion by bluprints-todd** [abpframework/abp#7844] [General]
Hi,

I am working on a new project and I want to use apb.io. I am a huge fan of the framework and I have implemented a ton of the older AspNetBoilerplate stuff.

For this project, I have a requirement where there are essentially 2 tiers **Customer -> Project**. Customers will sign up to the platform, and they can create one or more projects. Projects need to be independent of other projects (i.e: settings, features, users, roles, permissions, etc.). However, the one catch is that users in this system can be linked to multiple projects across different tenants.

 What would be the recommended approach given these requirements? So far I have considered the following:
- **Making the "Project" be the tenant** instead of the customer. This would allow for independent settings, features, users, roles, etc. In this scenario, I would need to manage the customer level manually and add new tables to support it.
- Keep the customer as the tenant, create a "Project" entity and build out some custom modules to support settings, features, users, roles, etc. at that level.
- Alternatively,  I could **turn off multi-tenancy** and manage everything manually.

Also, is there any documentation on the Identity module, specifically around the **"IdentityLinkUser"** entity and how that works? This seems like it might help me with the "users in this system can be linked to multiple projects across different tenants" requirements.

I am open to other suggestions or ideas and any guidance would be appreciated!

**Comment by cristiandmonda** [comment]
Hi, it's been a while since you posted this, what did you end up doing?
